File: app/src/main/java/com/prey/util/HttpUtil.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2019 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.util;

import android.os.StrictMode;

import com.prey.PreyLogger;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.URLConnection;

import java.net.URL;

public class HttpUtil {

    public static String getContents(String url) {
        String contents ="";
        InputStream in=null;
        try {
            StrictMode.ThreadPolicy policy =null;
            policy=new StrictMode.ThreadPolicy.Builder().permitAll().build();
            StrictMode.setThreadPolicy(policy);
            URLConnection conn = new URL(url).openConnection();
            in = conn.getInputStream();
            contents = convertStreamToString(in);
        } catch (Exception e) {
            PreyLogger.e("getContents error:"+e.getMessage(),e);
        } finally {
            if(in!=null){
                try {
                    in.close();
                } catch (Exception e) {
                }
            }
        }
        return contents;
    }

    private static String convertStreamToString(InputStream is) {
        InputStreamReader inReader=null;
        BufferedReader reader =null;
        StringBuilder sb =null;
        try {
            inReader=new InputStreamReader(is, "UTF-8");
            reader = new BufferedReader(inReader);
            sb = new StringBuilder();
            String line = null;
            while ((line = reader.readLine()) != null) {
                sb.append(line + "\n");
            }
        } catch (Exception e) {
        } finally {
            try {
                if(reader!=null)
                    reader.close();
            } catch (IOException e) {
            }
            try {
                if(inReader!=null)
                    inReader.close();
            } catch (IOException e) {
            }
            try {
                if(is!=null)
                    is.close();
            } catch (IOException e) {
            }
        }
        return sb==null?"":sb.toString();
    }

}

File: app/src/main/java/com/prey/util/HttpUtil.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.util

import android.os.StrictMode
import android.os.StrictMode.ThreadPolicy

import com.prey.PreyLogger

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.net.URL

/**
 * A utility class for making HTTP requests and converting input streams to strings.
 */
class HttpUtil {

    /**
     * Retrieves the contents of a URL as a string.
     *
     * @param url the URL to retrieve contents from
     * @return the contents of the URL as a string
     */
    fun getContents(url: String): String {
        var contents = ""
        var inputStream: InputStream? = null
        try {
            val policy = ThreadPolicy.Builder().permitAll().build()
            StrictMode.setThreadPolicy(policy)
            val connection = URL(url).openConnection()
            inputStream = connection.getInputStream()
            contents = convertStreamToString(inputStream)
        } catch (e: Exception) {
            PreyLogger.e("getContents error:" + e.message, e)
        } finally {
            if (inputStream != null) {
                try {
                    inputStream.close()
                } catch (e: Exception) {
                }
            }
        }
        return contents
    }

    /**
     * Converts an input stream to a string.
     *
     * @param inputStream the input stream to convert
     * @return the converted string
     */
    private fun convertStreamToString(inputStream: InputStream?): String {
        var inReader: InputStreamReader? = null
        var reader: BufferedReader? = null
        var sb: StringBuilder? = null
        try {
            inReader = InputStreamReader(inputStream, "UTF-8")
            reader = BufferedReader(inReader)
            sb = StringBuilder()
            var line: String? = null
            while ((reader.readLine().also { line = it }) != null) {
                sb.append(line + "\n")
            }
        } catch (e: Exception) {
        } finally {
            try {
                reader?.close()
            } catch (e: IOException) {
            }
            try {
                inReader?.close()
            } catch (e: IOException) {
            }
            try {
                inputStream?.close()
            } catch (e: IOException) {
            }
        }
        return sb?.toString() ?: ""
    }

}